configbootstrap_http.go

```diff
@@ -103,37 +103,63 @@ func configBootstrapHttp_bootstrapOne(
 }
 
 func (w ConfigBootstrapHttp) Bootstrap(ctx context.Context) (*ParsedConfig, string, error) {
-	if len(w.endpoints) == 1 {
-		return configBootstrapHttp_bootstrapOne(
-			ctx,
-			w.httpRoundTripper,
-			w.endpoints[0],
-			w.userAgent,
-			w.authenticator,
-			w.bucketName,
-		)
-	}
+	for {
```

Comment:
We should probably clean this up so that it's just 1 case (rather than having an `if len() == 1` block).  I also think that perhaps we should consider having configBootstrapHttp_bootstrapOne consider an empty vbucket map as something that it retries internally rather than having it return a parsedConfig that's invalid?